phylo/src/evolutionary_models/mod.rs

```diff
@@ -61,34 +79,55 @@ impl Display for ProteinModelType {
     }
 }
 
-pub trait EvolutionaryModelParameters<T> {
-    fn new(model_type: &T, params: &[f64]) -> Result<Self>
+impl ProteinModelType {
+    pub fn get_model_type(model_name: &str) -> Self {
+        match model_name.to_uppercase().as_str() {
+            "WAG" => ProteinModelType::WAG,
+            "BLOSUM" => ProteinModelType::BLOSUM,
+            "HIVB" => ProteinModelType::HIVB,
+            _ => {
+                warn!("Unknown DNA model requested, defaulting to WAG.");
+                ProteinModelType::WAG
+            }
+        }
+    }
+}
+
+pub trait EvoModelParams {
+    type ModelType;
+    type Parameter;
+    fn new(model: &Self::ModelType, params: &[f64]) -> Result<Self>
     where
         Self: Sized;
-    fn get_value(&self, param_name: &Parameter) -> f64;
-    fn set_value(&mut self, param_name: &Parameter, value: f64);
+    fn parameter_definition(
```

Comment:
It is absolutely not clear what is this, and what's the `static str` in the resulting vector. I think this is not correctly modeled, but I don't really know the logic.

Having static string smells, like having an enum, which is more type safe. 

If you want to have something printable, it absolutely make sense to add an enum (as suggested above) and having a formatting function, that translates that enum to a string for output.

Reply:
I will either refactor this in the next PRs or I will add a docstring. Can't think of how to do this better at the moment (also see comment below).